src/main.ts

```diff
@@ -0,0 +1,137 @@
+import {NotificationMessage} from './NotificationMessage';
+import {EventType} from "./enums";
+import {schemaOnIssue, schemaOnIssueComment, schemaOnPullRequest, schemaOnPush} from "./eventPayloadSchemaBuilder";
+
+const core = require ('@actions/core');
+const path = require('path');
+const https = require('https');
+
+const jsonPath = path.join(__dirname, '..', 'resources', 'notification.json');
+const workflow = process.env.GITHUB_WORKFLOW;
+const repository = process.env.GITHUB_REPOSITORY;
+const branch = process.env.GITHUB_REF;
+const github_event_payload =  process.env.GITHUB_EVENT_PATH || jsonPath;
+const trigger_event_name = process.env.GITHUB_EVENT_NAME;
+
+const msteams_webhook_url = core.getInput('webhook_url');
+const job_status = core.getInput('job_status');
+const event_id = core.getInput('event_id');
+
+export const parsedNotificationMessage = async function parse_event_to_message(event_payload_text: string): Promise<NotificationMessage> {
+    let details;
+    let account;
+    let message;
+    let url;
+    let parsedSchema;
+    try {
+        if (job_status.toUpperCase() === "CANCELLED") {
+            console.log("Job was cancelled, no notification will be sent")
+            process.exit(0);
+        }
+
+        let event_name : string = trigger_event_name || "event";
+        if (event_id) {
+            event_name = event_id;
+        }
+
+        switch(event_name){
+            case EventType.PUSH:
+                await schemaOnPush.validate(event_payload_text);
```

Comment:
It seems like yup's validate function returns the parsed value if it is valid, so I think it is not necessary to have a call to validate _and_ a call to cast.

Reply:
I see now that the yup validate function is async and rejects with ValidationErrors if not valid, so we _are_ at least catching errors. The question remains, what does that mean for the return type of the function?